### PR TITLE
Switch from GCC 5 that doesn't work with Eigen to 4.8

### LINKED ISSUE #3399: gcc version in CI machine is still 5.4
Although we install gcc-4.8 g++-4.8, but in nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04 image, gcc-5.4 and g++5.4 has been already installed. 
So, we must make gcc soft links to gcc4.8 instead of gcc5.4

### PR DESCRIPTION
**Pull request by QiJune**
fix #3399

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -71,6 +71,18 @@ RUN pip install --upgrade pip && \
 RUN apt-get install -y libssl-dev libffi-dev
 RUN pip install certifi urllib3[secure]
 
+# ln -sf to gcc4.8
+RUN ln -sf gcc-4.8 /usr/bin/gcc && \
```
Comment (wangkuiyi) [inline Dockerfile]: I think we need to remove gcc-5 in addition to linking to gcc-4.8:

```dockerfile
# The template library Eigen doesn't work well with GCC 5 coming with
# the default Docker image, so we remove GCC 5 and links to GCC 4.8.
apt purge gcc-5
```

File: paddle/operators/gaussian_random_op.cu [hunk 2]
```diff
@@ -33,7 +33,8 @@ class GaussianRandomKernel : public framework::OpKernel {
 
     int seed = context.op_.GetAttr<int>("seed");
     if (seed == 0) {
-      seed = std::random_device()();
+      std::random_device rd;
```
Comment (wangkuiyi) [inline paddle/operators/gaussian_random_op.cu]: The old way looks good to me. Do we really need this change?

**Comment by wangkuiyi** [comment]
I'd rename this PR as "Remove GCC 5 that doesn't work well with Eigen".

File: paddle/operators/gaussian_random_op.cu (hunk 2 above)
Comment (QiJune) [inline paddle/operators/gaussian_random_op.cu]: The old way will cause 'type name' declared as function returning a function error

**Pull request merged**